src/bootstrap/src/core/config/config.rs

```diff
@@ -2888,6 +2888,13 @@ impl Config {
 
         let absolute_path = self.src.join(relative_path);
 
+        // NOTE: This check is required because `jj git clone` doesn't create directories for
+        // submodules, they are completely ignored. The code below assumes this directory exists,
+        // so create it here.
+        if !absolute_path.exists() {
+            t!(fs::create_dir_all(&absolute_path));
+        }
```

Comment:
Remark: `Path::exists` will traverse symlinks to look for existence of the destination file (i.e. not necessarily the "shallow" `$checkout_root/$usual_submodule_location`), but I guess if the user has a symlink in the place of the usual submodules in checkout...